snapcraft/cli/env.py

```diff
@@ -15,41 +15,38 @@
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 import os
 from distutils import util
-from snapcraft.internal import errors, lxd
 
-
-class ContainerConfig:
-
-    def __init__(self):
-        """
-        Determines if a container should be used and which remote to choose
-
-        Checks the environment variable SNAPCRAFT_CONTAINER_BUILDS:
-          1. SNAPCRAFT_CONTAINER_BUILDS=1 enables local containers
-          2. SNAPCRAFT_CONTAINER_BUILDS=foobar uses the value as a remote
-          3. SNAPCRAFT_CONTAINER_BUILDS=0 or unset, no container is used
-        """
-
-        container_builds = os.environ.get('SNAPCRAFT_CONTAINER_BUILDS', '0')
-        # Default remote if it's a truthy value - otherwise it's a remote name
-        try:
-            self._use_container = util.strtobool(container_builds)
-            self._remote = None
-        except ValueError:
-            self._use_container = True
-            # Verbatim name of a remote
-            if not lxd._remote_is_valid(container_builds):
-                raise errors.InvalidContainerRemoteError(container_builds)
-            self._remote = container_builds
-
-    @property
-    def use_container(self):
-        return self._use_container
-
-    @property
-    def remote(self):
-        return self._remote
-
-
-def get_container_config():
-    return ContainerConfig()
+from . import echo
+from snapcraft.internal import errors
+
+
+class BuilderEnvironmentConfig:
+    """Handle the chosen build provider."""
+
+    def __init__(self) -> None:
+        use_lxd = None
+        container_builds = os.environ.get('SNAPCRAFT_CONTAINER_BUILDS')
+        if container_builds:
+            echo.warning(
+                'The flag SNAPCRAFT_CONTAINER_BUILDS has been deprecated. '
+                'Use SNAPCRAFT_BUILD_ENVIRONMENT=lxd instead.')
+            try:
+                use_lxd = util.strtobool(container_builds)
+            except ValueError:
+                raise errors.SnapcraftEnvironmentError(
+                    'The experimental feature of using lxd remotes other '
+                    'that cleanbuild has been dropped.')
```

Comment:
That phrasing makes little sense to me. Cleanbuild is a command. It supports LXD remotes via `--remote=`. What is actually being dropped here is support with non-local LXD remotes for persistent containers.
How about `The experimental feature of using non-local LXD remotes with SNAPCRAFT_CONTAINER_BUILDS has been dropped.`?